Assets/Scripts/Framework/StateMachine/States/GameState.cs

```diff
@@ -1,10 +1,48 @@
+using System.Collections;
+using FishNet;
+
+using DerailedDeliveries.Framework.Gameplay.Timer;
+using FishNet.Transporting;
+
 namespace DerailedDeliveries.Framework.StateMachine.States
 {
     /// <summary>
     /// The state that represents the game.
     /// </summary>
     public class GameState : MenuState
     {
+        public override IEnumerator OnStateEnter()
+        {
+            if (TimerUpdater.Instance.IsServer)
+                TimerUpdater.Instance.OnTimerCompleted += StopConnection;
+            else
+                InstanceFinder.ClientManager.OnClientConnectionState += ReturnToMenuState;
+
+            return base.OnStateEnter();
+        }
+
+        public override IEnumerator OnStateExit()
```

Comment:
Missing summary.

Reply:
I'll add summaries for this new method, the previous OnStateEnter that was already in that class was also in there without a summary and was also a overridden method so it didn't occur to me.